bot.js

```diff
@@ -219,6 +221,26 @@ client.on("guildMemberUpdate", memberUpdateHandler);
 // Every hour check if the current time is within the ranges
 setInterval(() => {
   const shouldPost = isRedditPostingTime();
+  if (!shouldPost) return;
+    Phin({ url: REDDIT_URL, parse: "json" })
+      .then((res) => {
+          redditPost = 0;
```

Comment:
What is this variable? you're defining it globally? Definitely not how you should keep track of data here.